hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -1181,23 +1239,23 @@ QString XmlChangeset::getChangeset(ChangesetInfoPtr changeset, long id, Changese
   if (changeset->size(ElementType::Node, type) > 0 || changeset->size(ElementType::Way, type) > 0 || changeset->size(ElementType::Relation, type) > 0)
   {
     ts << "\t<" << category << ">\n";
-    //  Nodes got first in each category
-    for (ElementIdToIdMap::iterator it = _idMap.begin(ElementType::Node); it != _idMap.end(ElementType::Node); ++it)
-    {
-      if (nodes.find(it->second) != nodes.end() && changeset->contains(ElementType::Node, type, it->second))
-        ts << nodes.at(it->second)->toString(id);
-    }
-    //  Followed by ways
-    for (ElementIdToIdMap::iterator it = _idMap.begin(ElementType::Way); it != _idMap.end(ElementType::Way); ++it)
+    if (type != ChangesetType::TypeDelete)
     {
-      if (ways.find(it->second) != ways.end() && changeset->contains(ElementType::Way, type, it->second))
-        ts << ways.at(it->second)->toString(id);
+      //  Nodes go first in each category
+      getNodes(changeset, ts, type, changeset_id);
+      //  Followed by ways
+      getWays(changeset, ts, type, changeset_id);
+      //  Relations bring up the rear
+      getRelations(changeset, ts, type, changeset_id);
```

Comment:
Order for adds and modifies is nodes, ways, relations and below the deletes begin with relations, ways, and then nodes.